block_test.go

```diff
@@ -1065,6 +1065,78 @@ func TestFencedCodeBlock(t *testing.T) {
 	doTestsBlock(t, tests, EXTENSION_FENCED_CODE)
 }
 
+func TestFencedCodeInsideBlockquotes(t *testing.T) {
+	var tests = []string{
+		"> ```go\n" +
+			"package moo\n\n" +
+			"```\n",
+		`<blockquote>
+<pre><code class="language-go">package moo
+
+</code></pre>
+</blockquote>
+`,
+		// -------------------------------------------
+		"> foo\n" +
+			"> \n" +
+			"> ```go\n" +
+			"package moo\n" +
+			"```\n" +
+			"> \n" +
+			"> goo.\n",
+		`<blockquote>
+<p>foo</p>
+
+<pre><code class="language-go">package moo
+</code></pre>
+
+<p>goo.</p>
+</blockquote>
+`,
+		// -------------------------------------------
+		"> foo\n" +
+			"> \n" +
+			"> quote\n" +
+			"continues\n" +
+			"```\n",
+		"<blockquote>\n" +
+			"<p>foo</p>\n\n" +
+			"<p>quote\n" +
+			"continues\n" +
+			"```</p>\n" +
+			"</blockquote>\n",
+
+		"> foo\n" +
+			"> \n" +
+			"> ```go\n" +
+			"package moo\n" +
+			"```\n" +
+			"> \n" +
+			"> goo.\n" +
+			"> \n" +
+			"> ```go\n" +
+			"package zoo\n" +
+			"```\n" +
+			"> \n" +
+			"> woo.\n",
+		`<blockquote>
+<p>foo</p>
+
+<pre><code class="language-go">package moo
+</code></pre>
+
+<p>goo.</p>
+
+<pre><code class="language-go">package zoo
+</code></pre>
+
+<p>woo.</p>
+</blockquote>
+`,
+	}
+	doTestsBlock(t, tests, EXTENSION_FENCED_CODE)
```

Comment:
I find these tests really hard to read.

I will propose an alternative way to represent those things in a PR (into this PR), which I find slightly more readable. I hope you will find it more readable too, then please take it.

I also think it's important to cover both forms of quoted fenced blocks and make sure they correctly preserve leading space. I think the current tests don't cover the 2nd form (but I could be mistaken, because it's really hard to read the test above):

	> plain quoted text
	> ```fenced
	code
	 with leading single space correctly preserved
	okay
	```
	> rest of quoted text

---

	> plain quoted text
	> ```fenced
	> code
	>  with leading single space correctly preserved
	> okay
	> ```
	> rest of quoted text

Both those inputs should generate (and they correctly do):

![image](https://cloud.githubusercontent.com/assets/1924134/10865861/b39e2edc-7fd7-11e5-93ab-bb5e8214547e.png)